File: src/main/java/com/ptah/repository/userprofiling/AuthorityMappingRepository.java
package com.ptah.repository.userprofiling;

import com.ptah.entity.userprofiling.AuthorityMapping;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;
import java.util.Set;

public interface AuthorityMappingRepository extends JpaRepository<AuthorityMapping,Long> {
     @Query("SELECT u FROM AuthorityMapping u WHERE u.role IN :roles")
     Set<AuthorityMapping> findByRoles(Set<String> roles);
}

File: backend/src/main/kotlin/com/ptah/repository/userprofiling/AuthorityMappingRepository.kt
package com.ptah.repository.userprofiling

import com.ptah.entity.userprofiling.AuthorityMapping
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface AuthorityMappingRepository : JpaRepository<AuthorityMapping?, Long?> {
    @Query("SELECT u FROM AuthorityMapping u WHERE u.role IN :roles")
    fun findByRoles(roles: Set<String?>?): Set<AuthorityMapping?>
    fun findByRole(role: String): AuthorityMapping?
}
